File: src/com/anathema_roguelike/entities/events/EnvironmentChangedEvent.java
package com.anathema_roguelike.entities.events;


import java.util.Optional;

import com.anathema_roguelike.environment.Environment;

public class EnvironmentChangedEvent {
	
	private Optional<Environment> oldEnvironment;
	private Environment newEnvironment;
	
	public EnvironmentChangedEvent(Optional<Environment> oldEnvironment, Environment newEnvironment) {
		this.oldEnvironment = oldEnvironment;
		this.newEnvironment = newEnvironment;
	}
	
	public Environment getNewEnvironment() {
		return newEnvironment;
	}
	
	public Optional<Environment> getOldEnvironment() {
		return oldEnvironment;
	}
	
	public void reRegister(Object o) {
		getOldEnvironment().ifPresent(old -> old.getEventBus().unregister(o));
		getNewEnvironment().getEventBus().register(o);
	}
}

File: src/com/anathema_roguelike/entities/events/EnvironmentChangedEvent.scala
package com.anathema_roguelike
package entities.events

import com.anathema_roguelike.environment.Environment

class EnvironmentChangedEvent(var oldEnvironment: Option[Environment], var newEnvironment: Environment) {
  def getNewEnvironment: Environment = newEnvironment

  def getOldEnvironment: Option[Environment] = oldEnvironment

  def reRegister(o: Any): Unit = {
    getOldEnvironment.foreach(old => old.getEventBus.unregister(o))
    getNewEnvironment.getEventBus.register(o)
  }
}
